src/components/measurements/index.tsx

```diff
@@ -131,39 +139,28 @@ const filterMeasurements = (
   filteredMeasurements: Measurement[]
 } => {
   // Find active filters to filter measurements
-  const activeFilters: {string?: string[]} = {};
-  Object.entries(filters).forEach(([field, valuesMap]) => {
-    activeFilters[field] = activeFilters[field] || [];
-    valuesMap.forEach(({active}, fieldValue) => {
-      // Save array of active values for the field filter
-      if (active) activeFilters[field].push(fieldValue);
-    });
-  });
+  const activeFilters = getActiveMeasurementFilters(filters);
 
   return {
     activeFilters,
     filteredMeasurements: measurements.filter((measurement) => {
       // First check the strain is visible in the tree
       if (!isVisible(treeStrainVisibility[measurement.strain])) return false;
       // Then check that the measurement contains values for all active filters
-      for (const [field, values] of Object.entries(activeFilters)) {
-        const measurementValue = measurement[field];
-        if (values.length > 0 &&
-           ((typeof measurementValue === "string") && !values.includes(measurementValue))){
-          return false;
-        }
-      }
-      return true;
+      return matchesAllActiveFilters(measurement, activeFilters);
     })
   };
 };
 
 const MeasurementsPlot = ({height, width, showLegend, setPanelTitle}) => {
+  const dispatch = useDispatch();
   // Use `lodash.isEqual` to deep compare object states to prevent unnecessary re-renderings of the component
   const { treeStrainVisibility, treeStrainColors } = useSelector((state: RootState) => treeStrainPropertySelector(state), isEqual);
-  const legendValues = useSelector((state: RootState) => state.controls.colorScale.legendValues, isEqual);
+  // Convert legendValues to string to ensure that subsequent attribute matches work as intended
+  const legendValues = useSelector((state: RootState) => state.controls.colorScale.legendValues.map((v): string => v.toString()), isEqual);
   const colorings = useSelector((state: RootState) => state.metadata.colorings);
   const colorBy = useSelector((state: RootState) => state.controls.colorBy);
```

Comment:
> What if clicking on the row again toggled back to the previously selected color?

I _think_ if we save the `colorBy` as a local ref to reference the previously selected color then it can be used to toggle the color back. If this gets too complicated, I might just toggle to the tree's default coloring.

Reply:
Added in 27cdbb07087c30e02184d3d4dc4e6d93dd8fb543